### PR TITLE
Fix(investigation main status): made structure changes

### PR DESCRIPTION
**Pull request by MayaSasson**
We have codes for main investigations' statues, so we changed the fk to use the id and not the name.
Because of that I changed some of the queries and mutations, and in addition created the statuses' code enum,
that we will be used as the indication for checking the status instead of the names enum.

### CODE REVIEW

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationStatusColumn/InvestigationStatusColumn.tsx [hunk 1]
```diff
@@ -8,17 +8,19 @@ import { Info, LockOpen } from '@material-ui/icons';
 import axios from 'Utils/axios';
 import { Severity } from 'models/Logger';
 import useCustomSwal from 'commons/CustomSwal/useCustomSwal';
-import InvestigationMainStatus from 'models/enums/InvestigationMainStatus';
+import InvestigationMainStatus from 'models/InvestigationMainStatus';
 
 import useStyles from './InvestigationStatusColumnStyles';
+import InvestigationMainStatusCodes from 'models/enums/InvestigationMainStatusCodes';
```
Comment (roymaniv) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationStatusColumn/InvestigationStatusColumn.tsx:14]: imports

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx [hunk 2]
```diff
@@ -350,13 +351,13 @@ const InvestigationTable: React.FC = (): JSX.Element => {
             case TableHeadersNames.investigationStatus:
                 const investigationStatus = indexedRow[cellName as keyof typeof TableHeadersNames];
                 const epidemiologyNumber = indexedRow[TableHeadersNames.epidemiologyNumber];
-                return <InvestigationStatusColumn
+                return investigationStatus ? <InvestigationStatusColumn
```
Comment (roymaniv) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx:354]: use && syntax

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTablesHeaders.ts [hunk 3]
```diff
@@ -1,4 +1,6 @@
+import InvestigationMainStatusCodes from 'models/enums/InvestigationMainStatusCodes';
 import SortOrder from 'models/enums/SortOrder';
+import InvestigationMainStatus from 'models/InvestigationMainStatus';
```
Comment (roymaniv) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTablesHeaders.ts:3]: imports

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts [hunk 4]
```diff
@@ -45,12 +45,13 @@ const getFlooredRandomNumber = (min: number, max: number): number => (
     Math.floor(Math.random() * (max - min) + min)
 )
 
+
```
Comment (roymaniv) [inline client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:48]: delete row

File: client/src/models/InvestigationMainStatus.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+interface InvestigationMainStatus  {
+    id: number,
+    displayName: string
```
Comment (roymaniv) [inline client/src/models/InvestigationMainStatus.ts:3]: ;

File: client/src/models/InvestigationMainStatus.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+}
+
+export default InvestigationMainStatus ;
```
Comment (roymaniv) [inline client/src/models/InvestigationMainStatus.ts:6]: remove space

File: client/src/redux/Status/statusesActionTypes.ts [hunk 7]
```diff
@@ -1,8 +1,10 @@
+import InvestigationMainStatus from "models/InvestigationMainStatus";
```
Comment (roymaniv) [inline client/src/redux/Status/statusesActionTypes.ts:1]: single quotes

File: server/src/Models/InvestigationStatus/GetAllInvestigationStatuses.ts [hunk 8]
```diff
@@ -1,13 +1,11 @@
+import InvestigationStatus from "./InvestigationMainStatus";
```
Comment (roymaniv) [inline server/src/Models/InvestigationStatus/GetAllInvestigationStatuses.ts:1]: single quotes

File: server/src/Models/InvestigationStatus/InvestigationMainStatus.ts [hunk 9]
```diff
@@ -0,0 +1,6 @@
+interface InvestigationMainStatus {
+    id: number,
+    displayName: string
```
Comment (roymaniv) [inline server/src/Models/InvestigationStatus/InvestigationMainStatus.ts:3]: ;

**Review by roymaniv — DISMISSED** [review]

**Review by EmilyLip — APPROVED** [review]

File: server/src/ClientToDBAPI/InvestigationInfo/mainRoute.ts [hunk 10]
```diff
@@ -183,11 +183,17 @@ investigationInfo.get('/subStatuses/:parentStatus', (request: Request, response:
         workflow: 'GraphQL GET subStatuses request to the DB',
     });
     graphqlRequest(GET_SUB_STATUSES_BY_STATUS, response.locals, {
-        parentStatus: request.params.parentStatus
+        parentStatus: +request.params.parentStatus
     })
         .then((result: any) => {
-            subStatusesLogger.info('GraphQL GET subStatuses request to the DB', Severity.LOW);
-            response.send(result)
+            if (result?.data?.allInvestigationSubStatuses?.nodes) {
```
Comment (ItayBenMoshe) [inline server/src/ClientToDBAPI/InvestigationInfo/mainRoute.ts:189]: don't you want to check !result?.errors as well

**Review by ItayBenMoshe — APPROVED** [review]

**Pull request merged**